src/charm.py

```diff
@@ -167,19 +292,40 @@ def _generate_spring_boot_layer(self) -> dict:
             },
         }
 
-    def _service_reconciliation(self) -> None:
-        """Run the reconciliation process for pebble services.
+    def _get_sprint_boot_container_memory_constraint(self) -> typing.Optional[int]:
+        """Get the spring-boot-app container memory limit.
+
+        Return:
+            The memory limit of the spring-boot-app container in number of bytes. ``None`` if
+            there's no limit.
 
         Raises:
-            ReconciliationError: unrecoverable errors happen during the Java application type
-                detection process, requiring the main reconciliation process to terminate the
-                reconciliation early.
+            RuntimeError: Container spring-boot-app does not exist, it shouldn't happen since
+                this function checks the existence of the spring-boot-app first.
         """
+        # ensure that the Spring Boot container is up
+        self._spring_boot_container()
+        kubernetes.config.load_incluster_config()
+        client = kubernetes.client.CoreV1Api()
+        spec: kubernetes.client.V1PodSpec = client.read_namespaced_pod(
```

Comment:
Wouldn't it be better to parse the statefullset instead ?

Reply:
Updated, thanks!

Reply:
There were some permission issues accessing StatefulSet using in-cluster authorization. I have reverted the change.